src/libraries/System.Runtime/tests/System.Reflection.Tests/AssemblyTests.cs

```diff
@@ -181,6 +183,32 @@ public void GetEntryAssembly()
             Assert.True(correct, $"Unexpected assembly name {assembly}");
         }
 
+        [Fact]
+        public void SetEntryAssembly()
+        {
+            Assert.NotNull(Assembly.GetEntryAssembly());
+
+            RemoteExecutor.Invoke(() =>
+            {
+                Assembly.SetEntryAssembly(null);
+                Assert.Null(Assembly.GetEntryAssembly());
+
+                Assembly originalAssembly = typeof(AssemblyTests).Assembly;
+
+                Assembly.SetEntryAssembly(originalAssembly);
+                Assert.Equal(Assembly.GetEntryAssembly(), originalAssembly);
```

Comment:
```suggestion
                Assembly testAssembly = typeof(AssemblyTests).Assembly;

                Assembly.SetEntryAssembly(testAssembly testAssembly);
                Assert.Equal(Assembly.GetEntryAssembly(), testAssembly);
```
This is not the original assembly

Reply:
I'm confused. If this is not the original assembly, then which is? Can you explain more in detail about this change?

Reply:
I would expect that the original value of `Assembly.GetEntryAssembly()` to be `Microsoft.DotNet.RemoteExecutor`, not the test assembly.

Reply:
I think you're right, that makes sense. I forgot that we were now running within the remote executor. Thanks for clarifying Jan